monkey/common/credentials/credentials.py

```diff
@@ -2,13 +2,34 @@
 
 from typing import Optional, Union
 
+from pydantic import SecretBytes, SecretStr
+
 from ..base_models import InfectionMonkeyBaseModel
 from . import LMHash, NTHash, Password, SSHKeypair, Username
 
 Secret = Union[Password, LMHash, NTHash, SSHKeypair]
 Identity = Username
 
 
+def get_plain_text(secret: Union[SecretStr, SecretBytes, None, str]) -> Optional[str]:
+    if secret:
+        return secret.get_secret_value()
+    else:
+        return secret
+
```

Comment:
If secret is a string that contains characters, this condition will evaluate to `True` and line 16 will raise an `AttributeError`

Maybe use an `isinstance()` check, or try/except. Make sure to add a unit test.

```suggestion
def get_plain_text(secret: Union[SecretStr, SecretBytes, None, str]) -> Optional[str]:
    try:
        return secret.get_secret_value()
    except AttributeError:
        return secret

```